source/server/listener_manager_impl.cc

```diff
@@ -211,6 +211,13 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, const std::st
       destination_ips.push_back(cidr_range.asString());
     }
 
+    // Validate source IP addresses.
+    std::vector<std::string> source_ips;
+    for (const auto& source_ip : filter_chain_match.source_prefix_ranges()) {
+      const auto& cidr_range = Network::Address::CidrRange::create(source_ip);
+      source_ips.push_back(cidr_range.asString());
```

Comment:
Why are we saving these to a vector?

Reply:
This is carryover from how it is implemented in [prefix_ranges](https://github.com/envoyproxy/envoy/blob/9c492a01db235d5f3b2871adff906784f789b8be/source/server/listener_manager_impl.cc#L209) . Is there a more suitable type that I should go with?

Reply:
I guess it's hard to see how things will be used without a fuller PR, doesn't seem unreasonable..